src/primary_beam.c

```diff
@@ -151,40 +151,97 @@ void vmCalcB(
+                if (errInt != 0)
+                {             
+                    handle_hyperbeam_error(__FILE__, (__LINE__ - 5), "fee_calc_jones");   
+                    exit(EXIT_FAILURE);
                 }
-#endif
-                mult2x2d_CxR( configs[config_idx], P, configs[config_idx] );
-#ifdef DEBUG
-                if (config_idx == 0)
+                
+                // Check elements and emit a warning if there are NaNs in the primary beam output.
+                // If NaNs are detected, set the Jones matrix to zero.
+                if (
+                    isnan(tempJones[0]) || 
+                    isnan(tempJones[1]) || 
+                    isnan(tempJones[2]) || 
+                    isnan(tempJones[3]) || 
+                    isnan(tempJones[4]) || 
+                    isnan(tempJones[5]) || 
+                    isnan(tempJones[6]) || 
+                    isnan(tempJones[7]) 
+                    )
                 {
-                    fprintf( stderr, "B       = " ); fprintf_complex_matrix( stderr, configs[config_idx] );
+                    printf("NaNs in Primary Beam Jones matrix!!?\n");
+                    memset( tempJones, 0, JONES_SIZE_BYTES );
                 }
-#endif
+                configs[config_idx] = (gpuDoubleComplex *)(tempJones);
             }
-
-            if (errInt !=0)
-            {             
-                handle_hyperbeam_error("Primary Beam",183, "fee_calc_jones");   
-                exit(EXIT_FAILURE);
+                
+            // Apply the parallactic angle correction, regardless of configuration
+#ifdef DEBUG
+            if (config_idx == 0)
+            {
+                fprintf( stderr, "Bhb     = " ); fprintf_complex_matrix( stderr, configs[config_idx] );
+                fprintf( stderr, "Ppa     = " ); fprintf( stderr, "[ %lf, %lf; %lf, %lf ]\n", P[0], P[1], P[2], P[3]  );
+             }
+#endif
+             mult2x2d_CxR( configs[config_idx], P, configs[config_idx] );
+#ifdef DEBUG
+            if (config_idx == 0)
+            {
+                fprintf( stderr, "B       = " ); fprintf_complex_matrix( stderr, configs[config_idx] );
             }
+#endif
 
             // Copy the answer into the B matrix (for this antenna)
+            // Since we do an operation on the matrix in this case, check again for NaNs
+            if (
+                isnan(configs[config_idx][0].x) || 
+                isnan(configs[config_idx][1].x) || 
+                isnan(configs[config_idx][2].x) || 
+                isnan(configs[config_idx][3].x) || 
+                isnan(configs[config_idx][0].y) || 
+                isnan(configs[config_idx][1].y) || 
+                isnan(configs[config_idx][2].y) || 
+                isnan(configs[config_idx][3].y)
+                )
+            {
+                printf("NaNs in Primary Beam Jones matrix after parallactic corrections!!?\n");
+                memset( configs[config_idx], 0, JONES_SIZE_BYTES );
```

Comment:
Actually is this okay? Or do we need to `memset` to `0` and then re-typecast?

Reply:
I'm not 100% sure what the question means, but let me know if this answers it:

It's fine. `memset` is "low-level" enough that it takes in a pointer of any kind and just dumps the value you give it at that location in memory. It doesn't care about what "type" of pointer you give it, and it also doesn't affect the type of that pointer in any way.

Reply:
This does indeed answer my question. I was 85% sure it was fine, but given I'm only a mediocre C guy, thought I'd check. Thanks!